dump/plugins/portchannel.py

```diff
@@ -0,0 +1,130 @@
+
+class PortChannel(Executor):
+    """
+    Debug Dump Plugin for PortChannel/LAG Module
+    """
+    ARG_NAME = "portchannel_name"
+    
+    def __init__(self):
+        self.match_engine = MatchEngine()
+        self.ret_temp = {}
+        self.ns = ''
+        self.lag_members = set()
+          
+    def get_all_args(self, ns=""):
+        req = MatchRequest(db="CONFIG_DB", table="PORTCHANNEL", key_pattern="*", ns=ns)
+        ret = self.match_engine.fetch(req)
+        all_lags = ret["keys"]
+        return [key.split("|")[-1] for key in all_lags]
+    
+    def execute(self, params_dict):
+        self.ret_temp = create_template_dict(dbs=["CONFIG_DB", "APPL_DB", "ASIC_DB", "STATE_DB"])
+        self.lag_name = params_dict[PortChannel.ARG_NAME]
+        self.ns = params_dict["namespace"]
+        # CONFIG_DB 
+        lag_found = self.init_lag_config_info()
+        if lag_found:
+            self.init_lag_member_config_info()
+        # APPL_DB
+        lag_found_appl = self.init_lag_appl_info()
+        if lag_found or lag_found_appl:
+            self.init_lag_member_appl_info()
+        # STATE_DB
+        lag_found_state = self.init_lag_state_info()
+        if lag_found or lag_found_appl or lag_found_state:
+            self.init_lag_member_state_info()
+        # ASIC_DB
+        lag_type_objs_asic = self.init_lag_member_type_obj_asic_info()
+        self.init_lag_asic_info(lag_type_objs_asic)
+        return self.ret_temp
+    
+    def add_to_ret_template(self, table, db, keys, err):
+        if not err and keys:
+            self.ret_temp[db]["keys"].extend(keys)
+            return True
+        else:
+            self.ret_temp[db]["tables_not_found"].extend([table])
+            return False
+    
+    def init_lag_config_info(self):
+        req = MatchRequest(db="CONFIG_DB", table="PORTCHANNEL", key_pattern=self.lag_name, ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        return self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"])
+        
+    def init_lag_member_config_info(self):
+        req = MatchRequest(db="CONFIG_DB", table="PORTCHANNEL_MEMBER", key_pattern=self.lag_name+"|*", ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        for key in ret["keys"]:
+            self.lag_members.add(key.split("|")[-1])
+        self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"])
+    
+    def init_lag_appl_info(self):
+        req = MatchRequest(db="APPL_DB", table="LAG_TABLE", key_pattern=self.lag_name, ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        return self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"])
+    
+    def init_lag_member_appl_info(self):
+        req = MatchRequest(db="APPL_DB", table="LAG_MEMBER_TABLE", key_pattern=self.lag_name+":*", ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        for key in ret["keys"]:
+            self.lag_members.add(key.split(":")[-1])
```

Comment:
Are APP_DB, STATE_DB and CONFIG_DB keys added to same lag_members structure?

Reply:
No, keys are added to the `self.ret_temp`, which is used in the top level debug command. 

`self.lag_members` is just used to capture the associated LAG members as seen in each of the tables, which is then used to figure out the LAG ASIC obj